crates/typst/src/introspection/counter.rs

```diff
@@ -372,6 +372,8 @@ impl Counter {
                     FigureElem::numbering_in(styles).clone()
                 } else if func == EquationElem::elem() {
                     EquationElem::numbering_in(styles).clone()
+                } else if func == FootnoteElem::elem() {
+                    FootnoteElem::numbering_in(styles).clone().into()
```

Comment:
I think an explicit `Some(..)` would be clearer than `.into()` here. I stumbled for a moment about what we're casting here.